File: 08/Java/TripleExpression/OpBigInteger.java
package TripleExpression;

import java.math.BigInteger;

public class OpBigInteger implements Operator<BigInteger> {
    public BigInteger parse(String a) {
        return new BigInteger(a);
    }

    public BigInteger add(BigInteger a, BigInteger b) throws Exception {
        return a.add(b);
    }
    public BigInteger subtract(BigInteger a, BigInteger b) throws Exception {
        return a.subtract(b);
    }
    public BigInteger multiply(BigInteger a, BigInteger b) throws Exception {
        return a.multiply(b);
    }
    public BigInteger divide(BigInteger a, BigInteger b) throws Exception {
        return a.divide(b);
    }
    public BigInteger negate(BigInteger a) throws Exception {
        return a.negate();
    }
    public BigInteger mod(BigInteger a, BigInteger b) throws Exception {
        return a.remainder(b);
    }
    public BigInteger square(BigInteger a) {
        return a.multiply(a);
    }
    public BigInteger abs(BigInteger a) {
        return a.abs();
    }

}

File: 08/CSharp/TripleExpressionApp/TripleExpressionApp/OpBigInteger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Numerics;

namespace TripleExpressionApp
{
    public class OpBigInteger : IOperator<BigInteger>
    {
        public BigInteger Parse(String a)
        {
            return new BigInteger();
        }

        public BigInteger Add(BigInteger a, BigInteger b)
        {
            return BigInteger.Add(a, b);
        }
        public BigInteger Subtract(BigInteger a, BigInteger b)
        {
            return BigInteger.Subtract(a, b);
        }
        public BigInteger Multiply(BigInteger a, BigInteger b)
        {
            return BigInteger.Multiply(a, b);
        }
        public BigInteger Divide(BigInteger a, BigInteger b)
        {
            return BigInteger.Divide(a, b);
        }
        public BigInteger Negate(BigInteger a)
        {
            return BigInteger.Negate(a);
        }
        public BigInteger Mod(BigInteger a, BigInteger b)
        {
            return BigInteger.Remainder(a, b);
        }
        public BigInteger Square(BigInteger a)
        {
            return BigInteger.Pow(a, 2);
        }
        public BigInteger Abs(BigInteger a)
        {
            return BigInteger.Abs(a);
        }
    }
}
